autoload/lsp/client.vim

```diff
@@ -82,7 +82,14 @@ function! s:on_stdout(id, data, event) abort
         if exists('l:response')
             " call appropriate callbacks
             let l:on_notification_data = { 'response': l:response }
-            if has_key(l:response, 'id')
+            if has_key(l:response, 'method') && has_key(l:response, 'id')
+                " it is a request from a server
+                let l:request = l:response
+                if l:request['method'] ==# 'workspace/applyEdit'
+                    call lsp#ui#vim#apply_workspace_edits(l:request['params']['edit'])
```

Comment:
we shouldn't call this directly here. can we instead expose a method similar to `on_notification` of `lsp#client#start`.

https://github.com/prabirshrestha/vim-lsp/blob/ee1dd1638a69930fb49be7f7e4e17c9bdb16de63/autoload/lsp.vim#L353

Reply:
Extracted to separate `on_request` callback similar to `on_notification`.